### DISCUSSION TITLE
Frigate+ training

### DISCUSSION THREAD

**Discussion by jakeycrx** [blakeblackshear/frigate#12516] [General]
The instructions for Frigate+ say to make sure you label everything in the image.  The problem is that there are always cars in my cameras field of view and it becomes very tedious labelling all the cars every single time.  My model picks up cars very well with scores into the mid 90's, so it doesn't really need any more training for cars.

My model needs a lot more training for cats and dogs, but it's a slow process labelling all the cars when I'm only interested in the cat or dog.  Will it dilute the model for cars if I don't label the cars?

Here is an example:

![image](https://github.com/user-attachments/assets/961a952a-75aa-4c65-bcb8-2e9c3b248831)

**Comment by NickM-27** [comment]
> Will it dilute the model for cars if I don't label the cars?

it will confuse the model, because you will be telling the training that those are not cars, when they in-fact are cars. There will be more tools including automatic labeling in the future to make this process easier

**Reply by jakeycrx** [reply]
Fair enough.  A mask or crop feature would be good for scenarios like this.  

Could I crop the images and manually feed them into Frigate+? Or does the image size need to be consistent?

**Reply by NickM-27** [reply]
part of the logic integral to the way frigate+ works is it deciding which parts of the image to user for training, as a user cropping or masking the images is just adding complication that won't give a benefit and has potential to degrade performance.

**Reply by NickM-27** [reply]
Once automatic labeling is implemented I don't think this should really be an issue

**Reply by jakeycrx** [reply]
> Once automatic labeling is implemented I don't think this should really be an issue

Do you have any idea of a rough timescale?

**Reply by NickM-27** [reply]
I don't work on frigate+ only frigate itself so I can't answer that, maybe @blakeblackshear can

**Reply by blakeblackshear** [reply]
I'm experimenting with label suggestions now. I think it's critical to have before expanding the available labels personally.

I have also been considering a way to mask out parts of the camera frame so it's blacked out during training. Some users already do with with a privacy screen built-in to the camera. I don't think that would hurt anything.

**Reply by jakeycrx** [reply]
A masking tool would be very useful.  Would it hurt anything if I were to mask out the cars using drawing software, then feed the images back into Frigate+? I guess black is the best colour?

**Reply by jakeycrx** [reply]
I just realised this won't really work because there is no option to say an object is not a cat.  Would it be possible to add that as an option as well?  I keep getting false positives for the same 3 items, so it would be good to label all 3 as not a cat in every image.

![image](https://github.com/user-attachments/assets/4bd3d6b9-8458-4302-9b6b-7b984567adec)
![image](https://github.com/user-attachments/assets/a7a556ab-4276-44dc-8aab-8f37584fa11e)

**Reply by NickM-27** [reply]
> so it would be good to label all 3 as not a cat in every image.

From what I understand that is done on purpose, users can not submit their own false positives because when a false positive is submitted it includes specific metadata about when it occurred and with which model, that way it can be reproduced and tested to see if future models reproduce it.

**Reply by bagobones** [reply]
> I have also been considering a way to mask out parts of the camera frame so it's blacked out during training. Some users already do with with a privacy screen built-in to the camera. I don't think that would hurt anything.

Instead of submitting the full camera frame, and masking for training why not just have the option to submit the cropped to region version that are used in the snapshots api / where previously presented as previews for detections? If necessary remove any zooming so they are original resolution just auto cropped? Would eliminate 90% of additional labeling needed when JUST submitting false positives or bounding box corrections.
